### PR TITLE
change hyperdiffusion in 3d test cases

### PR DESCRIPTION
**Pull request by szy21**
Use weak operators on the outside in the hyperdiffusion. cc @simonbyrne

### CODE REVIEW

File: examples/3dsphere/baroclinic_wave.jl [hunk 1]
```diff
@@ -205,7 +205,7 @@ function rhs!(dY, Y, _, t)
     χe = @. dρe = hwdiv(hgrad(cρe / cρ))
     χuₕ = @. duₕ =
         hwgrad(hdiv(cuₕ)) - Geometry.Covariant12Vector(
-            hwcurl(Geometry.Covariant3Vector(hcurl(cuₕ))),
+            hwcurl((hcurl(cuₕ))),
```
Comment (szy21) [inline examples/3dsphere/baroclinic_wave.jl:208]: @simonbyrne is it ok to change this?

**Comment by szy21** [comment]
bors try

**Comment by szy21** [comment]
bors try

File: examples/3dsphere/baroclinic_wave.jl [hunk 2]
```diff
@@ -204,20 +204,15 @@ function rhs!(dY, Y, _, t)
 
     χe = @. dρe = hwdiv(hgrad(cρe / cρ))
     χuₕ = @. duₕ =
-        hwgrad(hdiv(cuₕ)) - Geometry.Covariant12Vector(
-            hwcurl(Geometry.Covariant3Vector(hcurl(cuₕ))),
-        )
+        hwgrad(hdiv(cuₕ)) - Geometry.Covariant12Vector(hwcurl(hcurl(cuₕ)))
```
Comment (szy21) [inline examples/3dsphere/baroclinic_wave.jl:207]: @simonbyrne is it ok to change this?

Comment (simonbyrne) [reply examples/3dsphere/baroclinic_wave.jl:207]: It looks right to me.

**Comment by szy21** [comment]
bors try

**Comment by szy21** [comment]
@simonbyrne any idea why we can't use weak operators [here](https://github.com/CliMA/ClimaCore.jl/blob/1b59a5efe81623dcc051fe9d4feb400ff457201a/examples/hybrid/bubble_invariant.jl#L142)? See the error message [here](https://buildkite.com/clima/climacore-ci/builds/583#98a99562-fad6-4555-bffe-59aff1d3cfd9/96-105).

**Comment by szy21** [comment]
bors try

**Review by jiahe23 — APPROVED** [review]
🔥

**Pull request merged**